okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java

```diff
@@ -486,13 +487,23 @@ void onException(Throwable failureCause) {
    */
   private void onError(ErrorCode errorCode, String moreDetail) {
     frameWriter.goAway(0, errorCode, new byte[0]);
-    onGoAway(0, toGrpcStatus(errorCode).augmentDescription(moreDetail));
+    startGoAway(0, toGrpcStatus(errorCode).augmentDescription(moreDetail));
   }
 
-  private void onGoAway(int lastKnownStreamId, Status status) {
-    boolean notifyShutdown;
+  private void startGoAway(int lastKnownStreamId, Status status) {
+    synchronized (lock) {
+      if (goingAway) {
+        // Another go-away is inprocess, ignore this one.
```

Comment:
In my opinion, the `goAway` process lasts until the transport is finally shut down, during the process, the transport is `going away`, and after that, we don't care since the transport is fully stopped.

I don't think renaming the `goingAway`, `goAway`, `goAwayStatus` to `isShutingdown`, `isShutdown`, `shutdownStatus` will gain us any benefit, especially when `shutdown` implicates `stop per the caller's request`.